### PR TITLE
feat: configurable polling and ws

### PR DESCRIPTION
**Pull request by janek26**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

##

### CODE REVIEW

File: src/services/oracle.rs [hunk 1]
```diff
@@ -3,90 +3,265 @@ use std::{collections::HashMap, time::Duration};
...
     }
 
-    /// Update all the monitored assets with their latest USD price asynchronously.
-    async fn update_prices(&self) -> Result<()> {
+    async fn start_websocket(self) -> Result<()> {
+        tracing::info!("[🔮 Oracle] Updating prices via HTTP before starting WebSocket...");
+        self.update_prices_http().await?;
+
+        tracing::info!("[🔮 Oracle] Starting WebSocket connection...");
+        let ws_url = format!("wss://ws.{}/node/v1/data/subscribe", self.oracle.api_base);
```
Comment (azurwastaken) [inline src/services/oracle.rs:81]: this endpoint is for Starkex price, use this one instead wss://ws.dev.pragma.build/node/v1/data/price/subscribe

File: src/services/oracle.rs [hunk 2]
```diff
@@ -3,90 +3,265 @@ use std::{collections::HashMap, time::Duration};
...
+            )
+            .unwrap_or_default();
+
+            let re = Regex::new(r"/?usd$").unwrap();
+            let asset = re.replace(&asset_pair.to_lowercase(), "").into_owned();
+
+            if !asset.is_empty() {
+                if let Some(asset_info) = prices.get_mut(&asset) {
+                    let mut raw_price = oracle_price.median_price;
+                    let decimal_places = 18 - asset_info.decimals as usize;
```
Comment (azurwastaken) [inline src/services/oracle.rs:162]: Just for info, we have all the decimals in the config.yaml in project root as websocket doesn't return any decimals

File: src/types/account.rs [hunk 3]
```diff
@@ -58,7 +58,7 @@ impl StarknetAccount {
     /// Simulate a set of TXs and return the estimation of the fee necessary
     /// to execute them.
     pub async fn estimate_fees_cost(&self, txs: &[Call]) -> Result<BigDecimal> {
-        let estimation = self.0.execute_v1(txs.to_vec()).estimate_fee().await?;
+        let estimation = self.0.execute_v3(txs.to_vec()).estimate_fee().await?;
```
Comment (azurwastaken) [inline src/types/account.rs:61]: why switch to v3 ?

File: src/types/account.rs [hunk 4]
```diff
@@ -67,7 +67,7 @@ impl StarknetAccount {
 
     /// Executes a set of transactions and returns the transaction hash.
     pub async fn execute_txs(&self, txs: &[Call]) -> Result<Felt> {
-        let res = self.0.execute_v1(txs.to_vec()).send().await?;
+        let res = self.0.execute_v3(txs.to_vec()).send().await?;
```
Comment (azurwastaken) [inline src/types/account.rs:70]: same here

File: src/cli/mod.rs [hunk 5]
```diff
@@ -93,6 +160,10 @@ impl RunCmd {
                 }
             }
         }
+
+        println!("⚙️ Configuration after validation:");
```
Comment (azurwastaken) [inline src/cli/mod.rs:164]: switch those two line to tracing

**Review by azurwastaken — COMMENTED** [review]
Thanks for your work ! just some smol comment

**Comment by azurwastaken** [comment]
also can you update the .env.example with the new values please ?

**Comment by janek26** [comment]
Hey! Spoke to 0xevolve that I dont have much time nowadays, but wanted to open this PR to show what might be nice to have to improve the bot. Feel free take as much inspiration as you want, but unfortunately I wont have time to update my branch for now ..

Re your semantics comments, I'm an absolute rust noob and most code is by Claude, which explains the "wrong" imports etc

Let me know if I can provide more context, but I dont think that I'll find time in the next weeks to update my fork with your comments, sorry about that ❤️

**Pull request closed without merging**